packages/site/src/layout/AnimatedPresenceDisclosure.tsx

```diff
@@ -1,29 +1,66 @@
 import { AnimatedPresence, Button, Typography } from "@jobber/components";
-import React, { useState } from "react";
-import { Link } from "react-router-dom";
+import React, { useEffect, useState } from "react";
+import { Link, useLocation } from "react-router-dom";
 import styles from "./NavMenu.module.css";
 
 interface AnimatedPresenceDisclosureProps {
   readonly children: React.ReactNode;
   readonly title: React.ReactNode;
   readonly to: string;
+  readonly selected?: boolean;
 }
 
 function AnimatedPresenceDisclosure({
   children,
   title,
   to,
+  selected,
 }: AnimatedPresenceDisclosureProps) {
-  const [isOpen, setIsOpen] = useState(false);
+  const location = useLocation();
+
+  // Determine if any child is selected based on the current URL
+  const hasSelectedChild = React.Children.toArray(children).some(
```

Comment:
Another minor improvement. Since we now have `React.Children.toArray(children)` twice in this component and I don't think that value will change we could pull the value into a memoized variable.

Reply:
Another great call out - did that here 867d9da1953700271dc57169bdb780a115717c7c